azurerm/resource_arm_kusto_database.go

```diff
@@ -0,0 +1,220 @@
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Kusto Database creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		server, err := client.Get(ctx, resourceGroup, clusterName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Kusto Database %q (Resource Group %q, Cluster %q): %s", name, resourceGroup, clusterName, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_kusto_database", *server.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+
+	databaseProperties := expandKustoDatabaseProperties(d)
+
+	kustoDatabase := kusto.Database{
+		Name:               &name,
+		Location:           &location,
+		DatabaseProperties: databaseProperties,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, clusterName, name, kustoDatabase)
+	if err != nil {
+		return fmt.Errorf("Error creating or updating Kusto Cluster %q (Resource Group %q, Cluster %q): %+v", name, resourceGroup, clusterName, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Kusto Cluster %q (Resource Group %q, Cluster %q): %+v", name, resourceGroup, clusterName, err)
+	}
+
+	resp, getDetailsErr := client.Get(ctx, resourceGroup, clusterName, name)
+	if getDetailsErr != nil {
+		return fmt.Errorf("Error retrieving Kusto Cluster %q (Resource Group %q, Cluster %q): %+v", name, resourceGroup, clusterName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read ID for Kusto Cluster %q (Resource Group %q, Cluster %q)", name, resourceGroup, clusterName)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmKustoDatabaseRead(d, meta)
+}
+
+func resourceArmKustoDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).kusto.DatabasesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	clusterName := id.Path["Clusters"]
+	name := id.Path["Databases"]
+
+	databaseResponse, err := client.Get(ctx, resourceGroup, clusterName, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(databaseResponse.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Kusto Database %q (Resource Group %q, Cluster %q): %+v", name, resourceGroup, clusterName, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("cluster_name", clusterName)
+
+	if location := databaseResponse.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	databaseProperties := *databaseResponse.DatabaseProperties
```

Comment:
We'll need to nil check this here:
```suggestion
	if props := databaseResponse.DatabaseProperties; props != nil {
```

Reply:
Added a nil check and merged master into it.